### PR TITLE
Add initial support for SiWx917

### PR DESCRIPTION
**Pull request by jerome-pouiller**
This PR allows to properly start UART driver. However, the quality is still far below my expectations. Many parameters are hard code and many aspects are still unclear.

This PR has been tested with Zephyr 3.7rc1 and a specific version of hal_silabs that include WiSeConnect 3.3.

### CODE REVIEW

**Review by asmellby — APPROVED** [review]
Do we want to require signoff on commits for DCO? https://docs.zephyrproject.org/latest/contribute/guidelines.html#developer-certification-of-origin-dco

Other than that, this looks good to me.

**Comment by jerome-pouiller** [comment]
v2:
  - Add link to hal_silabs
  - Add README
  - Add Signed-Off

File: README.md [hunk 1]
```diff
@@ -1,2 +1,28 @@
...
+-----
+
+Install `west` command. Usually, it is done through `pip` (or `pipx`):
+
+    pip install west
+
+Retrieve this repository:
+
+    mkdir workspace
+    cd worksapce
```
Comment (asmellby) [inline README.md:17]: Inconsequential typo.

**Review by asmellby — APPROVED** [review]

**Comment by jerome-pouiller** [comment]
I have just noticed `checkpatch.pl` raises many warnings on this PR.

**Review by jhedberg — APPROVED** [review]

**Pull request merged**